### PR TITLE
Java 8 recommendation [GEOS-9619]

### PR DESCRIPTION
**Pull request by jodygarnett**
Taking discussion from https://github.com/geoserver/geoserver/pull/4243 into a pull request with concrete changes.

## Checklist

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [x] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: doc/en/user/source/production/container.rst [hunk 1]
```diff
@@ -55,12 +55,12 @@ For more information about JVM configuration, see the article `Performance tunin
...
    The above results (from a 16 GB laptop) amount to initial heap size of 256m, and a max heap size of around 4 GB (or around 1/4 of system memory).
-   
+
+.. _production_container.marlin:
+
 Enable the Marlin rasterizer
 ----------------------------
 
-Before Java 9, OpenJDK and Oracle used the Pisces and Ductus renderers to rasterize vector data respectively.  In Java 9 onward they use Marlin which has better overall performance in most situations than either Pisces or Ductus.
-
```
Comment (jodygarnett) [inline doc/en/user/source/production/container.rst:63]: Used this into on the prior page where it talks about java versions.

File: doc/en/user/source/production/java.rst [hunk 2]
```diff
@@ -3,98 +3,125 @@
...
+   * - Java 6 
+     - 
+     - tested
+     -
+     - GeoServer 2.3.x to 2.5.x
+   * - Java 5 
+     - 
+     - 
+     - tested
+     - GeoServer 2.2.x and earlier
```
Comment (jodygarnett) [inline doc/en/user/source/production/java.rst:48]: While I do not think anyone is working on Java 5 anymore this is much easier to read as a table

File: doc/en/user/source/production/java.rst [hunk 3]
```diff
@@ -3,98 +3,125 @@
...
 
-As of GeoServer 2.0, a Java Runtime Environment (JRE) is sufficient to run GeoServer.  GeoServer no longer requires a Java Development Kit (JDK).
+As of GeoServer 2.0, a Java Runtime Environment (JRE) is sufficient to run GeoServer.  GeoServer no longer requires installation of a complete Java Development Kit (JDK) with compiler.
 
-Running on Java 11
-------------------
+Running on Java 8 (Recommended)
+-------------------------------
```
Comment (jodygarnett) [inline doc/en/user/source/production/java.rst:53]: Clear recommendation of Java 8, provides a summary of the options, and then a heading to explore each idea.

File: doc/en/user/source/production/java.rst [hunk 4]
```diff
@@ -196,3 +223,16 @@ Installing native JAI manually
 You can install the native JAI manually if you encounter problems using the above installers, or if you wish to install the native JAI for more than one JDK/JRE.
 
 Please refer to the `GeoTools page on JAI installation <http://docs.geotools.org/latest/userguide/build/install/jdk.html#java-extensions-optional>`_ for details.
+
+Running on Java 11
+------------------
```
Comment (jodygarnett) [inline doc/en/user/source/production/java.rst:228]: New heading for Java 11, provides a summary of options, this way it is very clear Java 11 works out of the box and does not have as many optional installs.

File: doc/en/user/source/production/java.rst [hunk 5]
```diff
@@ -3,98 +3,125 @@
...
+      "Strong cryptography is available"
 
-The policy files are available at
+.. note::
 
-* `IBM JCE policy jars <https://www14.software.ibm.com/webapp/iwm/web/preLogin.do?source=jcesdk>`_ 
+   The installing these policy files needs to be done for each update of the Java runtime. 
 
-An IBM ID is needed to log in. The installation is identical to Oracle.
+Native JAI and ImageIO extensions (not recommended)
```
Comment (jodygarnett) [inline doc/en/user/source/production/java.rst:117]: Clear Java 8 subheading for Native JAI and ImageIO, and clearly not recommended.

I considered taking this section to the wiki and removing it from our day-to-day user manual.

File: doc/en/user/source/production/java.rst [hunk 6]
```diff
@@ -196,3 +223,16 @@ Installing native JAI manually
...
+
+Running on Java 11
+------------------
+
+GeoServer is tested with Java 11 (LTS), with industry `support up to 2024 <https://adoptopenjdk.net/support.html#roadmap>`_. 
+
+GeoServer 2.15 onward will run under Java 11 with no additional configuration on **Tomcat 9** or newer and **Jetty 9.4.12** or newer. Running GeoServer using Java 11 on other Application Servers may require some additional configuration as not all Application Servers support Java 11 yet.
+
+* Java 11 already includes Marlin Renderer, although you may wish to consider installing a newer version
+* Java 11 includes Installing Unlimited Strength Jurisdiction Policy Files
```
Comment (aaime) [inline doc/en/user/source/production/java.rst:235]:
```suggestion
* Java 11 includes the Unlimited Strength Jurisdiction Policy Files, no need for a separate installation
```

**Pull request merged**